xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-rest/src/main/java/org/xwiki/livedata/internal/rest/DefaultLiveDataEntriesResource.java

```diff
@@ -66,25 +71,22 @@
     private static final String FILTERS_PREFIX = "filters.";
 
     @Override
-    public Entries getEntries(String sourceId, String namespace, List<String> properties, List<String> matchAll,
+    public Entries getEntries(String sourceId, String namespaceStr, List<String> properties, List<String> matchAll,
         List<String> sort, List<Boolean> descending, long offset, int limit) throws Exception
     {
-        // Workaround for https://github.com/restlet/restlet-framework-java/issues/922 (JaxRs multivalue query-params
-        // gives list with null element)
-        List<String> actualProperties = properties.stream().filter(Objects::nonNull).collect(Collectors.toList());
-        List<String> actualMatchAll = matchAll.stream().filter(Objects::nonNull).collect(Collectors.toList());
-        List<String> actualSort = sort.stream().filter(Objects::nonNull).collect(Collectors.toList());
-        List<Boolean> actualDescending = descending.stream().filter(Objects::nonNull).collect(Collectors.toList());
-        LiveDataConfiguration config =
-            getConfig(sourceId, actualProperties, actualMatchAll, actualSort, actualDescending, offset, limit);
-        Optional<LiveDataSource> source = this.liveDataSourceManager.get(config.getQuery().getSource(), namespace);
-        if (source.isPresent()) {
-            String idProperty = config.getMeta().getEntryDescriptor().getIdProperty();
-            LiveDataEntryStore entryStore = source.get().getEntries();
-            return createEntries(entryStore.get(config.getQuery()), idProperty, config.getQuery().getSource(),
-                namespace).withOffset(offset).withLimit(limit);
-        } else {
-            throw new WebApplicationException(Response.Status.NOT_FOUND);
+        XWikiContext xWikiContext = this.xcontextProvider.get();
```

Comment:
Did you try using the ``ModelContext`` to set the current entity instead (passing the wiki reference)? It would be a bit more cleaner I think.

Reply:
Indeed that looks cleaner, testing.

Reply:
Importing `ModelContext` requires to also import `WikiReference`, leading to a `ClassFanOutComplexity` checkstyle error. 
I'll see how it can refactor the code to avoid this issue.